### PR TITLE
Video listing

### PR DESCRIPTION
**Pull request by mnath30**
Added the following:
1. Created Video Listing page
2. Created 404 page 
3. Created Loader and error page on fetching the videos

### CODE REVIEW

File: src/components/VideoCard/VideoCard.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import "./video-card.css";
+import { useState } from "react";
+import { PopUpMenu } from "../PopUpMenu/PopUpMenu";
+
+const VideoCard = ({ videoDetails }) => {
+  const { title, creator, views, thumbnail, img } = videoDetails;
+  const [showMenu, setShowMenu] = useState(false);
+  const titleNew = title.length > 36 ? title.slice(0, 33) + "..." : title;
```
Comment (omjadhav85) [inline src/components/VideoCard/VideoCard.jsx:8]: Nice idea to shorten the title....i had used css ellipsis property for this but this seems easier

**Review by sharathnair9999 — APPROVED** [review]

File: src/pages/VideoListing/VideoListing.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  const { allVideos } = videoState;
+  const { loader, videoData, error } = useFetch("/api/videos");
+  useEffect(() => {
+    videoDispatch({ type: SET_VIDEOS, payload: videoData });
+  }, [videoData, videoDispatch]);
+
+  return (
+    <div className={`flex video__container {showMobileNav? "no-scroll":""}`}>
+      {loader && <Loader />}
+      {allVideos !== [] &&
```
Comment (omjadhav85) [inline src/pages/VideoListing/VideoListing.jsx]: better way could be to check allVideos's length like allVideos.length !== 0

**Review by omjadhav85 — APPROVED** [review]
Great work! Like the use of useFetch custom hook and the logic to shorten the title name. Some minor suggestions provided but its good to merge!!!

File: src/pages/VideoListing/VideoListing.jsx (hunk 2 above)
Comment (mnath30) [reply src/pages/VideoListing/VideoListing.jsx]: Thankyou for your comments. I have made the suggested changes

**Pull request merged**